Feature: Pure Go Lua (VM)

GlenDC opened the pull request:
This PR replaces the CGO Lua VM (https://github.com/siddontang/golua) with a Lua VM written in pure Golang (https://github.com/yuin/gopher-lua), and fixes #288 

Because of this the `lua` build tag is no longer needed, and neither is any of the CGO information related to the C Lua VM.

I also updated the documentation to reflect this PR, and the previous PR (#289).

I've split up this PR in 2 commits, the first one with the dependency changes (vendor) and the second one with the actual changes, this way you can more easily review this PR by simply focussing on the diff of the second commit.

siddontang commented:
Cool @GlenDC 

Do you use `make update_vendor` to re-update the dependency? seem that some needless files are introduced in vendor.

GlenDC commented:
@siddontang oh no sorry, totally missed that _Make_ target. Was manually updating the dependencies using the glide tool, and than copying them over to the `_vendor` folder. Will force update the PR, with the first commit replaced by a new commit.

GlenDC commented:
@siddontang you were totally right, and thanks for providing that make target, I wish I had seen that earlier, hehe. The dependency commit is now last (instead for first), and contains indeed quite some files less, than the original first commit. Good for pointing that out!

GlenDC commented:
@siddontang I also added support for CJSON, as that was also supported previously before. I think this PR is good to go now. Let me know what you think.

siddontang commented on server/cmd_script.go:
```diff
@@ -71,39 +67,48 @@ func evalGenericCommand(c *client, evalSha1 bool) error {
+	// makes a mistake using `ledis.call`
+	defer func() {
+		if r := recover(); r != nil {
+			switch x := r.(type) {
```
I think using `err = fmt.Errorf("panic: %v", r)` is enough.

siddontang commented:
Thanks @GlenDC 

Seem that Go Lua is more easy to use. 

Rest LGTM

GlenDC commented:
Thank you to @siddontang, for the fast responses, I really appreciate that.
I agree that using Gopher-Lua is quite easy, compared to the CGO lua.

I added and rebased the change you requested 👍
